components/messaging/ConversationPartner.tsx

```diff
@@ -0,0 +1,89 @@
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+import { getProfile } from '@/utils/supabase/getProfile';
+import { createSupabaseClient as supabase } from '../../utils/supabase/createSupabaseClient';
+import { useConversationContext } from '@/context/conversationContext';
+import { MessageType } from '@/types/messagingTypes';
+
+type ConversationPartnerProps = {
+  message_data: MessageType[];
+};
+
+type ConversationPartnerType = {
+  username: string;
+  avatar: string;
+};
+
+export const ConversationPartner: React.FC<ConversationPartnerProps> = ({
+  message_data,
+}) => {
+  const [conversationPartner, setConversationPartner] = useState<
+    ConversationPartnerType[] | undefined
+  >();
+  const { currentConversation } = useConversationContext();
+
+  useEffect(() => {
+    const getConversationPartner = async () => {
+      const conversationPartnerSet = new Set(
+        message_data?.map((message: { sender_id: string }) => message.sender_id)
+      );
+
+      if (
+        conversationPartnerSet?.size !== 0 &&
+        conversationPartnerSet?.size !== 1
+      ) {
+        const calculateID = (isCurrentUser: boolean) => {
+          return (
+            conversationPartnerSet &&
+            Array.from(conversationPartnerSet).find((id) =>
+              isCurrentUser
+                ? id === currentConversation?.user_id
+                : id !== currentConversation?.user_id
+            )
+          );
+        };
+
+        const conversationDonorID = calculateID(true);
+        const conversationPartnerID = calculateID(false);
+
+        let partnerProfile;
+
+        if (conversationPartnerID && conversationPartnerID !== undefined) {
+          partnerProfile = await getProfile(supabase, conversationPartnerID);
+        }
+
+        conversationDonorID &&
+          (await getProfile(supabase, conversationDonorID));
+
+        partnerProfile && setConversationPartner(partnerProfile.data);
+      } else {
+        const { data: fetchedItemDonor } = await supabase
+          .from('items')
+          .select('profiles(username, avatar)')
+          .eq('id', currentConversation?.item_id);
+
+        fetchedItemDonor &&
+          setConversationPartner(fetchedItemDonor[0].profiles);
+      }
+    };
+    getConversationPartner();
+  }, [currentConversation?.id]);
+
+  return (
+    <div className='flex flex-row items-center p-5'>
+      <p data-testid='item-donor'>
+        <b>From: </b>
+        {conversationPartner?.[0]?.username}
```

Comment:
The conversationPartnerSet has been striped to only one profile before you setConversationPartner so the value here is a single profile.

When I run this on my local the value for every conversation is nullish, but works if I remove the index specification.

`conversationPartner?.[0]?.username`
`conversationPartner?.[0]?.avatar` (on line 84)
![Screenshot 2024-03-18 at 14 58 40](https://github.com/enBloc-org/kindly/assets/114600712/097822f8-6753-4617-aaa9-9bfb8e7f95d7)

`conversationPartner?.username`
`conversationPartner?.avatar`
![Screenshot 2024-03-18 at 14 59 22](https://github.com/enBloc-org/kindly/assets/114600712/337126c4-0d8e-4664-aa4d-220b3c1d1d7b)

Reply:
Yep, that does it.

It looks like the issue is TypeScript is expecting that return value from Supabase to have each of the keys typed 'any' and when it sees our ConversationPartnerType asking for 'string' on each of the object keys it freaks out.

I tried deconstructing the return and assigning it manually within the setConversationPartner statement but for some reason that was leaving me with nulls as well in some cases.

I'm happy to go ahead with this assertion 🤷